src/game/frontend/submenus/Teleport.cpp

```diff
@@ -1,14 +1,200 @@
 #include "Teleport.hpp"
+
+#include "game/features/Features.hpp"
 #include "game/frontend/items/Items.hpp"
+#include "game/services/custom_teleport_service.hpp"
+#include "util/Math.hpp"
+#include "util/Teleport.hpp"
+#include "core/frontend/Notifications.hpp"
+#include "game/backend/FiberPool.hpp"
 
 namespace YimMenu::Submenus
 {
+	Telelocation GetLocationPlayerIsClosestTo()
+	{
+		if (!Pointers.GetLocalPed)
+			return {};
+
+		float distance = 500;
+		Telelocation closestLocation{};
+		//saved_locations_filtered_list can be used to get a joint list of all categories when the filter is empty.
+		for (auto& loc : g_CustomTeleportService.SavedLocationsFilteredList())
+		{
+			float newDistance = Math::DistanceBetweenVectors(Self::Pos, {loc.x, loc.y, loc.z});
+
+			if (newDistance < distance)
+				closestLocation = loc, distance = newDistance;
+		}
+
+		return closestLocation;
+	}
+
+	float GetDistanceToTelelocation(Telelocation t)
+	{
+		return Math::DistanceBetweenVectors(Vector3(t.x, t.y, t.z), Self::Pos);
+	}
+
+
+	void RenderCustomTeleport()
+	{
+		ImGui::BeginGroup();
+		static std::string NewLocationName{};
+		static std::string category = "Default";
+		static Telelocation deleteTelelocation;
+		static std::string filter{};
+
+		if (!std::string(deleteTelelocation.name).empty())
+			ImGui::OpenPopup("##deletelocation");
+
+		if (ImGui::BeginPopupModal("##deletelocation", nullptr, ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove))
+		{
+			ImGui::Text("Are you sure you want to delete %s?", deleteTelelocation.name);
+
+			ImGui::Spacing();
+
+			if (ImGui::Button("Yes"))
+			{
+				g_CustomTeleportService.DeleteSavedLocation(category, deleteTelelocation.name);
+				deleteTelelocation.name = "";
+				ImGui::CloseCurrentPopup();
+			}
+			ImGui::SameLine();
+			if (ImGui::Button("No"))
+			{
+				deleteTelelocation.name = "";
+				ImGui::CloseCurrentPopup();
+			}
+
+			ImGui::EndPopup();
+		}
+
+		ImGui::PushItemWidth(300);
+		InputTextWithHint("Category", "Category", &category).Draw();
+
+		InputTextWithHint("Location name", "New location", &NewLocationName).Draw();
+		ImGui::PopItemWidth();
+
+		if (ImGui::Button("Save current location")) //Button widget still crashes
```

Comment:
Button widget will crash because of the lambda capture only capturing [this] i believe. Try using [&] in the widget

Reply:
Still crashes